test/composableStableJoin.integration.test.ts

```diff
@@ -0,0 +1,387 @@
+        expect(expectedMaxAmountsIn).to.deep.eq(
+            maxAmountsIn.filter((_, index) => index !== bptIndex),
+        );
+    });
+
+    test('single asset join', async () => {
+        const bptIndex = txInput.poolInput.tokens.findIndex(
+            (t) => t.address === txInput.poolInput.address,
+        );
+        const bptOut = TokenAmount.fromHumanAmount(bptToken, '1');
+        const tokenIn = '0x4bc3263eb5bb2ef7ad9ab6fb68be80e43b43801f';
+
+        // perform join query to get expected bpt out
+        const joinInput: SingleAssetJoinInput = {
+            bptOut,
+            tokenIn,
+            chainId,
+            rpcUrl,
+            kind: JoinKind.SingleAsset,
+        };
+
+        const { queryResult, maxAmountsIn, minBptOut, value } =
+            await doTransaction({
+                ...txInput,
+                joinInput,
+            });
+        // Query should use same bpt out as user sets
+        expect(queryResult.bptOut.amount).to.deep.eq(bptOut.amount);
+
+        // We only expect single asset to have a value for amount in
+        expect(queryResult.tokenInIndex).toBeDefined;
+        queryResult.amountsIn
+            .filter((_, index) => index !== bptIndex)
+            .forEach((a, i) => {
+                if (i === queryResult.tokenInIndex)
+                    expect(a.amount > BigInt(0)).to.be.true;
+                else expect(a.amount === BigInt(0)).to.be.true;
+            });
+
+        // Should be no native value
+        expect(value).toBeUndefined;
+
+        // Confirm slippage - only to amount in not bpt out
+        const expectedMaxAmountsIn = queryResult.amountsIn.map((a) =>
+            txInput.slippage.applyTo(a.amount),
+        );
+        expect(expectedMaxAmountsIn).to.deep.eq(maxAmountsIn);
+        expect(minBptOut).to.eq(bptOut.amount);
+    });
+
+    test('proportional join', async () => {
+        const bptOut = TokenAmount.fromHumanAmount(bptToken, '1');
+        const bptIndex = txInput.poolInput.tokens.findIndex(
+            (t) => t.address === txInput.poolInput.address,
+        );
+
+        // perform join query to get expected bpt out
+        const joinInput: ProportionalJoinInput = {
+            bptOut,
+            chainId,
+            rpcUrl,
+            kind: JoinKind.Proportional,
+        };
+
+        const { queryResult, maxAmountsIn, minBptOut, value } =
+            await doTransaction({
+                ...txInput,
+                joinInput,
+            });
+
+        // Query should use same bpt out as user sets
+        expect(queryResult.bptOut.amount).to.deep.eq(bptOut.amount);
+
+        // Expect all assets to have a value for amount in
+        expect(queryResult.tokenInIndex).toBeDefined;
+        queryResult.amountsIn
+            .filter((_, index) => index !== bptIndex)
+            .forEach((a) => {
+                expect(a.amount > BigInt(0)).to.be.true;
+            });
+        expect(queryResult.tokenInIndex).toBeUndefined;
```

Comment:
Strange! Seems like `expect(queryResult.tokenInIndex === undefined).toEqual(true or false);` works.
We also have this in weightedJoin.integration.tests.ts and should also change any use of `toBeDefined` and `toBeUndefined`, this can be done in a separate PR and I'll make an Asana ticket.

Reply:
True, That's very odd, the tests shouldn't be working

Reply:
I've already done the weighted tests (you'll need to merge latest join) but we still need to do it here @lgahdl .

Reply:
We can also use "expect(value).to.be.undefined" or "expect(value).to.not.be.undefined", works fine